themes/bootstrap3/templates/RecordDriver/DefaultRecord/core.phtml

```diff
@@ -1,9 +1,14 @@
 <div class="media" vocab="http://schema.org/" resource="#record" typeof="<?=$this->driver->getSchemaOrgFormats()?> Product">
   <?php
+
     $QRCode = $this->record($this->driver)->getQRCode("core");
     $coverDetails = $this->record($this->driver)->getCoverDetails('core', 'medium', $this->record($this->driver)->getThumbnail('large'));
     $cover = $coverDetails['html'];
     $preview = $this->record($this->driver)->getPreviews();
+
+    //override the default preview link with overdrive link
+    if($this->OverdrivePreview)  $preview = $this->OverdrivePreview;
```

Comment:
Rather than putting Overdrive-specific language into this file, can we do this more generically? For example, call this `$this->previewOverride` and perhaps replace the `OverdriveAction` and `OverdriveAvail` with a `$this->extraControls` variable that is output if non-empty? That will make this template more reusable with other children without having to add lots of additional special cases.

Reply:
Yes. I like it.

Reply:
I changed my mind on this.  Although the previewOverride kind of makes sense, I can't find a good solution for the "extra controls" that aren't replacing something.  

One of the things I didn't like about our approach for the child template is that it's really hard for someone to override the template in their own theme.  I discovered that when I tried to do it myself.  If it's just on big block of text coming from the child, then you don't have much option for changing the layout.  By at least breaking the two Overdrive blocks (actions buttons and availability block) up, you can move each one around a bit.  If you want to actually change the markup within, it's even more of a pain (but at least you know where to go look for it).  If it's just a generic blob of markup coming from any of the children, how would you know might be in there?  For example, if you know you don't use overdrive, you know you can safely  comment that line out in your custom theme.  If you do and you want to modify the markup, you can try to follow it to the overdrive template and attempt to change it.  

Am I making any sense?  Maybe there is a middle ground...

Reply:
I guess we could do $this->actionControls and $this->availabilityInfo or something

Reply:
My main issue here is that it's not really helpful to have variables with the name "overdrive" in them, because that leads us down a road where we potentially have many different service-specific exceptions in the core template, which will make it hard to read and maintain. It also doesn't make sense to have tons of specific conditions like this, because we're only ever going to be dealing with one service at a time under normal circumstances. If it's valuable to you to be able to locally override the core template in a way that reflects Overdrive-specific behavior, I believe that you can check `$this->driver->setSourceIdentifier()` to identify when Overdrive is active without having to rely on specific named variables being set.

Also note that if you really want a totally different layout for Overdrive in your local custom version, you can copy the core template from DefaultRecord in the default theme into the Overdrive folder of your custom theme and adjust to your heart's content... then it will be loaded dynamically only when Overdrive record drivers are loaded. This may be more of a headache to maintain in the long term because you'll have to keep it in sync with the core template as that template evolves over time (which is why we generally try to avoid copying and pasting templates), but it's certainly manageable.

Does that address your concern at all, or am I missing the point?

Reply:
Ok so does using the more generic $this->actionControls and $this->availabilityInfo work for you?
or maybe this:
if(!empty($this->extraControls) {
  echo extraControls['actionControls'];
  echo extraControls['availabilityInfo'];
}

Reply:
Yes, that works for me. I'd maybe include some `?? ''` fallbacks on the ends of those echo statements so notices are not triggered if somebody sets one key but not the other.